dojo/tools/trivy/parser.py

```diff
@@ -0,0 +1,79 @@
+"""
+Parser for Aquasecurity trivy (https://github.com/aquasecurity/trivy) Docker images scaner
+"""
+
+import json
+
+from dojo.models import Finding
+
+
+TRIVY_SEVERITIES = {
+    "CRITICAL": "Critical",
+    "HIGH": "High",
+    "MEDIUM": "Medium",
+    "LOW": "Low",
+    "UNKNOWN": "Info",
+}
+
+DESCRIPTION_TEMPLATE = """{title}
+Target: {target}
+Fixed version: {fixed_version}
+
+{description_text}
+"""
+
+
+class TrivyParser:
+
+    def __init__(self, scan_file, test):
+        self.items = []
+
+        scan_data = scan_file.read()
+
+        try:
+            data = json.loads(str(scan_data, 'utf-8'))
+        except:
+            data = json.loads(scan_data)
+
+        if not isinstance(data, list):
+            return
+
+        for target_data in data:
+            if not isinstance(target_data, dict) or 'Target' not in target_data:
+                continue
+            target = target_data['Target']
+            vulnerabilities = target_data.get('Vulnerabilities', []) or []
+            for vuln in vulnerabilities:
+                if not isinstance(vuln, dict):
+                    continue
+                try:
+                    vuln_id = vuln['VulnerabilityID']
+                    package_name = vuln['PkgName']
+                    package_version = vuln['InstalledVersion']
+                    severity = TRIVY_SEVERITIES[vuln['Severity']]
+                except KeyError:
```

Comment:
If you're missing say key "InstalledVersion", the whole vulnerability will be skipped. Is this what you wanted to do? Also, I would probably add a log statement there, just in case instead of just being completely silent.

Reply:
Right,
> "Errors should never pass silently."

Fixed together with improved handling of the missing package version.